### PR TITLE
CMSIS-NN: Update average pool tc to new interface

### PR DESCRIPTION
**Pull request by felix-johnny**
1. Existing test case is updated as per new interface
2. Four new test cases are added

### CODE REVIEW

**Comment by felix-johnny** [comment]
@mansnils Please have a look at this PR as well.

**Comment by felix-johnny** [comment]
Depends on https://github.com/ARM-software/CMSIS_5/pull/931

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_avgpool_s8/test_arm_avgpool_s8.c [hunk 1]
```diff
@@ -16,36 +16,247 @@
...
+  pool_params.padding.w = AVGPOOLING_PAD_X;
+  pool_params.padding.h = AVGPOOLING_PAD_Y;
+  pool_params.stride.w = AVGPOOLING_STRIDE_X;
+  pool_params.stride.h = AVGPOOLING_STRIDE_Y;
+
+  pool_params.activation.min = AVGPOOLING_OUT_ACTIVATION_MIN;
+  pool_params.activation.max = AVGPOOLING_OUT_ACTIVATION_MAX;
+
+  ctx.size = arm_avgpool_s8_get_buffer_size(AVGPOOLING_INPUT_W, AVGPOOLING_IN_CH);
+  ;
```
Comment (mansnils) [inline CMSIS/NN/Tests/UnitTest/TestCases/test_arm_avgpool_s8/test_arm_avgpool_s8.c:60]: Abundant semicolon

File: CMSIS/NN/Tests/UnitTest/TestCases/TestData/avgpooling_1/config_data.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#pragma once
+// Generated by generate_test_data.py
+#define AVGPOOLING_1_OUT_CH 3
+#define AVGPOOLING_1_IN_CH 3
+#define AVGPOOLING_1_CH_MULT 1  // (Only for depthwise conv)
```
Comment (gmiodice) [inline CMSIS/NN/Tests/UnitTest/TestCases/TestData/avgpooling_1/config_data.h:24]: is it correct this comment in this file?

Comment (felix-johnny) [reply CMSIS/NN/Tests/UnitTest/TestCases/TestData/avgpooling_1/config_data.h:24]: @gmiodice The comment is auto generated as well :-)

Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/TestData/avgpooling_1/config_data.h:24]: Yeah we could improve the generate script in future version, so it just leave out the line completely unless it is for depthwise.

**Pull request merged**